crd-generator/api/src/main/java/io/fabric8/crd/generator/v1/decorator/AddAdditionPrinterColumnDecorator.java

```diff
@@ -40,41 +42,42 @@ public AddAdditionPrinterColumnDecorator(String name, String version, String typ
 
   @Override
   public void andThenVisit(CustomResourceDefinitionVersionFluent<?> spec) {
-    Predicate<CustomResourceColumnDefinitionBuilder> matchingColumn = col -> col.getName() != null && col.getName().equals(columnName);
+    Predicate<CustomResourceColumnDefinitionBuilder> matchingColumn = col -> col.getName() != null
+        && col.getName().equals(columnName) && col.getJsonPath() != null && col.getJsonPath().equals(path);
```

Comment:
Just highlighting the "real change" in this PR.

Reply:
Perhaps we should issue a warning when we encounter such a situation?

Reply:
I checked the behavior, and it looks consistent to me without the need for a warning.
Using the "Joke" CRD used in the tests and created a corresponding "Joke" CR:

```yaml
apiVersion: "samples.javaoperatorsdk.io/v1alpha1"
kind: JokeRequest
metadata:
  name: jr-any
spec:
  category: Any
status:
  category: Pun
```

the output of `kubectl get` seems fine to me:
![Screenshot 2022-11-29 at 10 47 21](https://user-images.githubusercontent.com/5792097/204509354-d83e7fc1-d8fb-4632-a766-31d8c1158a93.png)